CyLR/src/archive/SharpZipArchive.cs

```diff
@@ -10,11 +10,16 @@ class SharpZipArchive : Archive
     {
         private readonly ZipOutputStream archive;
 
-        public SharpZipArchive(Stream destination)
+        public SharpZipArchive(Stream destination, String password)
             : base(destination)
         {
             archive = new ZipOutputStream(destination);
             archive.IsStreamOwner = false;
+            if (password != null)
```

Comment:
using `string.IsNullOrEmpty(password)` here instead probably makes this a bit more fool-proof